File: ChatGPT/190_3a.py
import random

def getMonthlySales():
    # Simulate getting monthly sales as a random number between 0 and 100000
    return random.randint(0, 100000)

# Initialize a list to store monthly sales
monthly_sales = []

# Call getMonthlySales 12 times, once for each month
for _ in range(12):
    sales = getMonthlySales()
    monthly_sales.append(sales)

# Initialize quarterly sales
quarterly_sales = [0, 0, 0, 0]

# Calculate the sales for each quarter
for i in range(4):
    quarterly_sales[i] = sum(monthly_sales[i*3:(i+1)*3])

# Print the results
for i, sales in enumerate(quarterly_sales):
    print(f"Quarter {i + 1} Sales: ${sales}")

# Optional: Print monthly sales for reference
print("\nMonthly Sales:")
for i, sales in enumerate(monthly_sales):
    print(f"Month {i + 1}: ${sales}")

File: ChatGPT_Cpp/CWE-190/190_3a.cpp
#include <iostream>
#include <cstdlib>  // For rand() and srand()
#include <ctime>    // For time()

// Function to simulate getting monthly sales (returns a random number between 0 and 100000)
int getMonthlySales() {
    return rand() % 100001;  // Random number between 0 and 100000
}

int main() {
    // Seed random number generator
    srand(static_cast<unsigned int>(time(0)));

    int quarterlySales[4] = {0};  // Array to store the sales for each quarter

    // Call getMonthlySales 12 times (once for each month) and add up the quarterly sales
    for (int month = 0; month < 12; ++month) {
        int sales = getMonthlySales();
        std::cout << "Month " << (month + 1) << " sales: " << sales << std::endl;

        // Determine which quarter the month belongs to and add the sales to the corresponding quarter
        int quarter = month / 3;
        quarterlySales[quarter] += sales;
    }

    // Print the quarterly sales
    for (int q = 0; q < 4; ++q) {
        std::cout << "Quarter " << (q + 1) << " total sales: " << quarterlySales[q] << std::endl;
    }

    return 0;
}
